public/pages/ThreatIntel/utils/helpers.ts

```diff
@@ -0,0 +1,276 @@
+  scanConfig.per_ioc_type_scan_input_list.forEach(({ ioc_type, index_to_fields_map }) => {
+    Object.entries(index_to_fields_map).forEach(([index, fieldAliases]) => {
+      if (!logSourcesByName[index]) {
+        logSourcesByName[index] = {
+          name: index,
+          iocConfigMap: {
+            [ioc_type]: {
+              fieldAliases,
+              enabled: true,
+            },
+          },
+        };
+      } else {
+        logSourcesByName[index].iocConfigMap = {
+          ...logSourcesByName[index].iocConfigMap,
+          [ioc_type]: {
+            fieldAliases,
+            enabled: true,
+          },
+        };
+      }
+    });
+  });
+
+  const configClone: any = _.cloneDeep(scanConfig);
+  delete configClone.per_ioc_type_scan_input_list;
+
+  configClone.triggers.forEach((trigger: ThreatIntelAlertTrigger) => {
+    if (trigger.actions.length === 0) {
+      trigger.actions.push(getEmptyThreatIntelAlertTriggerAction());
+    }
+  });
+
+  const formModel: ThreatIntelScanConfigFormModel = {
+    ...configClone,
+    logSources: Object.values(logSourcesByName),
+  };
+
+  return formModel;
+}
+
+export function configFormModelToMonitorPayload(
+  formModel: ThreatIntelScanConfigFormModel
+): ThreatIntelMonitorPayload {
+  const fieldAliasesByIocType: { [k in ThreatIntelIocType]?: IocFieldAliases } = {};
+
+  formModel.logSources.forEach((source) => {
+    Object.entries(source.iocConfigMap).forEach(([iocType, iocConfig]) => {
+      if (!iocConfig?.enabled) {
+        return;
+      }
+
+      if (!fieldAliasesByIocType[iocType as ThreatIntelIocType]) {
+        fieldAliasesByIocType[iocType as ThreatIntelIocType] = {
+          ioc_type: iocType as ThreatIntelIocType,
+          index_to_fields_map: {
+            [source.name]: iocConfig.fieldAliases,
+          },
+        };
+      } else {
+        fieldAliasesByIocType[iocType as ThreatIntelIocType]!.index_to_fields_map[source.name] =
+          iocConfig.fieldAliases;
+      }
+    });
+  });
+
+  const formModelClone: any = _.cloneDeep(formModel);
+  delete formModelClone['logSources'];
+
+  return {
+    ...formModelClone,
+    per_ioc_type_scan_input_list: Object.values(fieldAliasesByIocType),
+  };
+}
+
+export function readIocsFromFile(
+  file: File,
+  onRead: (
+    readResponse: { ok: true; sourceData: FileUploadSource } | { ok: false; errorMessage: string }
+  ) => void
+) {
+  if (file.size > 512000) {
+    return onRead({ ok: false, errorMessage: 'File size should be less then 500KB.' });
+  }
+
+  const reader = new FileReader();
+  reader.readAsText(file);
+  reader.onload = function () {
+    try {
+      const iocStrings =
+        reader.result
+          ?.toString()
+          .split('\n')
+          .filter((iocObj) => !!iocObj) || [];
+      const iocs = [];
+      for (let iocStr of iocStrings) {
+        try {
+          const ioc = JSON.parse(iocStr);
+          if (typeof ioc !== 'object') {
+            throw '';
```

Comment:
let's throw some error

Reply:
Not needed because we have hardcoded message in catch below